src/plugins/vis_augmenter/public/utils/utils.ts

```diff
@@ -58,3 +64,38 @@ export const buildPipelineFromAugmentVisSavedObjs = (objs: ISavedAugmentVis[]):
     throw new Error('Expression function from augment-vis saved objects could not be generated');
   }
 };
+
+/**
+ * Returns an error with an aggregated message about all of the
+ * errors found in the set of VisLayers. If no errors, returns undefined.
+ */
+export const getAnyErrors = (visLayers: VisLayer[], visTitle: string): Error | undefined => {
+  const visLayersWithErrors = visLayers.filter((visLayer) => isVisLayerWithError(visLayer));
+  if (!isEmpty(visLayersWithErrors)) {
+    // Aggregate by unique plugin resource type
+    const resourceTypes = [
+      ...new Set(visLayersWithErrors.map((visLayer) => visLayer.pluginResource.type)),
+    ];
+
+    let msgDetails = '';
+    resourceTypes.forEach((type, index) => {
+      const matchingVisLayers = visLayersWithErrors.filter(
+        (visLayer) => visLayer.pluginResource.type === type
+      );
+      if (index !== 0) msgDetails += '\n\n\n';
```

Comment:
style nit, no need to change: this is basically a `join` operation on an array of strings - we could refactor this message composition so that each of the `forEach` loops are just mapping operations that return arrays of strings and then join them together to form the final message rather than just concatenating everything together as we go.